gnome-initial-setup/pages/live-chooser/eos-setup-live-user

```diff
@@ -178,6 +165,18 @@ class SetupLiveUser(object):
         self.update(GS_SCHEMA, ALLOW_UPDATES, GLib.Variant('b', False))
 
 
+def install_desktop_file():
+    """eos-installer is shipped in all images, but its desktop file is
+    (deliberately) only present in the GDM greeter's environment. For
+    image-booted live sessions, we symlink it into place."""
+    if not os.path.exists(LOCAL_DESKTOP_PATH):
+        log.info('ensuring %s exists', LOCAL_APPS_DIR)
+        os.makedirs(LOCAL_APPS_DIR, exist_ok=True)
+
+        log.info('symlinking %s to %s', EOS_INSTALLER, LOCAL_DESKTOP_PATH)
+        os.symlink(EOS_INSTALLER_PATH, LOCAL_DESKTOP_PATH)
```

Comment:
I think a more pythonic way of doing this would be:

    os.makedirs(...)
    try:
        os.symlink(EOS_INSTALLER_PATH, LOCAL_DESKTOP_PATH)
    except OSError as error:
        if error.errno != errno.EEXIST:
            raise error

`os.path.exists` is technically racy, though I appreciate that this is a somewhat naff point.